### PR TITLE
PR: Allow activate-environment to be a path

### PR DESCRIPTION
**Pull request by bollwyvl**
This follows up on #136 where we might want to allow for `activate-environment` to be a `--prefix` rather than a `--name` argument to various conda commands.
 
Also removes some dead code which was using `--name`, and tweaks some more docs and typos in log messages.

Of worrying note: I have now seen at least one `403` from the GitHub API when requesting the miniforge releases, despite my best efforts at caching. We probably need to:
- have a hard-coded "version of last resort" (the one where it dropped msys2 would be a good one :P) 
- throw some weight on https://github.com/conda-forge/miniforge/issues/76
- use `git` to list tags and build URLs from that... but we might pull pre-releases, etc.

### CODE REVIEW

**Comment by goanpeca** [comment]
@bollwyvl I did not understand the issues before, querying Gtihub API is expensive as you have now noticed.

Why do we need that?

**Comment by bollwyvl** [comment]
> Why do we need the miniforge releases?

At present, `miniforge-version` is not _required_, though of course we could: demanding a version seems like it might be less nice for the simplest use case. Unless we do, we need to get a version from _somewhere_ as there is no `latest` at a predictable URL. Having the page on the above linked issue would give us that, as would hard-coding a version-of-last-resort... it seems to be getting released more often than miniconda.

**Comment by goanpeca** [comment]
>At present, miniforge-version is not required, though of course we could: demanding a version seems like it might be less nice for the simplest use case.

Might be less nice, but is what you have to provide to overcome the unnecessary problems with the API and havig a page with all that (not happenning soon) so I prefer going with explicit over implicit here :)

**Comment by jaimergp** [comment]
This might not be as nice and convenient for the user but what if we deprecate `miniconda-version`, `miniforge-version`, `miniforge-variant` and friends and just expose `installer`? This could be either a URL to a constructor-derived installer, or a keyword like `Miniconda-4.12` or `Mambaforge-4.12`.

If finding the right installer URL is tricky, we could collect a list in our up and coming fancy docs.

**Comment by bollwyvl** [comment]
> rather have the miniforge-version requirement.

Sure, sounds good. seems out of scope for this PR, will do another... one thing we _can_ do then is have a default `miniforge-variant` of `Miniforge`, so it would still only be a single input to get "vanilla" conda/cpython at a particular version.

**Comment by goanpeca** [comment]
>Sure, sounds good. seems out of scope for this PR, will do another... one thing we can do then is have a default miniforge-variant of Miniforge, so it would still only be a single input to get "vanilla" conda/cpython at a particular version.

Agreed!

--- 

@jaimergp let's explore our options on another issue :-p

**Comment by goanpeca** [comment]
Is this ready for review @bollwyvl ?

**Comment by bollwyvl** [comment]
> deprecate miniconda-version, miniforge-version, miniforge-variant

i feel like deprecation makes people sad... because fixing ci that _used to work_ makes people sad. and the URL structures are fairly different between them, anyhow, even though they both come out of constructor. another reason i like the separate ones is that it becomes easy to use github search for [examples](
https://github.com/search?l=YAML&q=miniconda-version&type=Code) when you have semantically-rich strings.

**Comment by bollwyvl** [comment]
> Is this ready for review

yeah, unless we want to sneak anything else into the tests... i guess I haven't tried giving it a `base` env, might break the world...

**Comment by bollwyvl** [comment]
oh, and @goanpeca, re: #136: can you comment on why we always add `$CONDA/condabin` to path?

**Comment by bollwyvl** [comment]
Looks like we need to keep `condabin`... will revert...

File: src/conda.ts [hunk 1]
```diff
@@ -144,10 +157,7 @@ export async function condaInit(
   options: types.IDynamicOptions
 ): Promise<void> {
   let ownPath: string;
-  const isValidActivate: boolean =
-    inputs.activateEnvironment !== "base" &&
-    inputs.activateEnvironment !== "root" &&
-    inputs.activateEnvironment !== "";
+  const isValidActivate = !utils.isBaseEnv(inputs.activateEnvironment);
```
Comment (goanpeca) [inline src/conda.ts:160]: is `utils.isBaseEnv` also checking for an empt string env?

**Review by goanpeca — APPROVED** [review]
One question, otherwise looks good :) 

Thanks @bollwyvl !

File: src/conda.ts (hunk 1 above)
Comment (bollwyvl) [reply src/conda.ts:160]:
```

/** Names for a conda `base` env */
export const BASE_ENV_NAMES = ["root", "base", ""];
```

**Pull request merged**